### PR TITLE
Hugging Face Datasets

### PR DESCRIPTION
**Pull request by PaulHax**
Test it out:
`nrtk-explorer --dataset cppe-5 beans rafaelpadilla/coco2017 keremberke/german-traffic-sign-detection mrtoy/mobile-ui-design keremberke/construction-safety-object-detection keremberke/table-extraction`

Use the `--download` CLI arg to cache the Hugging Face dataset localy.  Dataset streaming is the default.
`nrtk-explorer --dataset cppe-5 --download`

Adds a `dataset[vision]` dependency.

Object Detection task tagged datasets:
https://huggingface.co/datasets?task_categories=task_categories:object-detection
Does not support them all =/

### CODE REVIEW

File: src/nrtk_explorer/library/dataset.py [hunk 1]
```diff
@@ -1,60 +1,213 @@
...
+    for identifier in dataset_identifiers:
+        if is_coco_dataset(identifier):
+            expanded_identifiers.append(identifier)
+        else:
+            infos = get_dataset_infos(identifier)
+            for config_name, info in infos.items():
+                for split_name in info.splits:
+                    streaming_str = "streaming" if streaming else "download"
+                    expanded_identifiers.append(
+                        f"{identifier}@{config_name}@{split_name}@{streaming_str}"
```
Comment (PaulHax) [inline src/nrtk_explorer/library/dataset.py:92]: This is bananas 🍌  Is there a better way\ to bring some "config" along with the hugging face dataset?

**Comment by PaulHax** [comment]
@Erotemic Have you ever tried shoehorning a Hugging Face dataset into a COCO shape?  Or any other image dataset format for that matter?

**Comment by Erotemic** [comment]
@PaulHax other data formats: yes. It is usually possible because the different image / video annotation formats are almost all doing the same thing in different ways. Some of them make more assumptions than others. I think I've done a reasonable job at giving kwcoco the representation power to handle almost everything.

Here are examples of code in kwcoco to handle and help convert to/from other formats: https://gitlab.kitware.com/computer-vision/kwcoco/-/tree/main/kwcoco/formats?ref_type=heads

Namely, I have code for:

* VOC
* LabelME
* KW18 (very old Kitware format)
* KPF (newer Kitware format: https://github.com/Kitware/DIVA/blob/master/doc/manuals/kpf.rst). 

On top of this, I've done fairly specific conversions for VIAME, which code is scattered around for: https://github.com/VIAME/bioharn/tree/main/dev/data_tools

There have also been different non-public formats that can be converted.

**Review by alesgenova — APPROVED** [review]
I tried this branch out and it works well!
I don't see anything wrong in the code, but I also don't have experience with kwcoco and hf

**Pull request merged**